### PR TITLE
test: kill warnings

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

File: src/test/librbd/test_internal.cc [hunk 1]
```diff
@@ -664,7 +664,7 @@ TEST_F(TestInternal, ResizeCopyup)
   bufferlist bl;
   bl.append(std::string(4096, '1'));
   for (size_t i = 0; i < m_image_size; i += bl.length()) {
-    ASSERT_EQ(bl.length(), ictx->io_work_queue->write(i, bl.length(),
+    ASSERT_EQ((ssize_t)bl.length(), ictx->io_work_queue->write(i, bl.length(),
                                                       bufferlist{bl}, 0));
```
Comment (dillaman) [inline src/test/librbd/test_internal.cc]: Nit: indentation is off here

File: src/test/librbd/test_internal.cc [hunk 2]
```diff
@@ -733,7 +733,7 @@ TEST_F(TestInternal, DiscardCopyup)
...
```
Comment (dillaman) [inline src/test/librbd/test_internal.cc]: Nit: indentation is off here

**Comment by tchaikov** [comment]
@dillaman fixed and repushed

File: src/test/librados_test_stub/MockTestMemIoCtxImpl.h [hunk 3]
```diff
@@ -106,6 +106,7 @@ class MockTestMemIoCtxImpl : public TestMemIoCtxImpl {
                      std::map<uint64_t, uint64_t> *m, bufferlist *bl){
      return TestMemIoCtxImpl::sparse_read(oid, off, len, m, bl);
   }
+  using TestMemIoCtxImpl::sparse_read;
```
Comment (dillaman) [inline src/test/librados_test_stub/MockTestMemIoCtxImpl.h]: The warning is that the ```size_t len``` parameter doesn't match -- the param type needs to be fixed and then you can removed this ```using```.

Comment (tchaikov) [inline src/test/librados_test_stub/MockTestMemIoCtxImpl.h]: thanks, fixed.

**Pull request merged**